tests/e2e/gc_rwx_reclaim_policy.go

```diff
@@ -0,0 +1,568 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"strings"
+
+	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	cnstypes "github.com/vmware/govmomi/cns/types"
+	v1 "k8s.io/api/core/v1"
+	storagev1 "k8s.io/api/storage/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	"k8s.io/kubernetes/test/e2e/framework"
+	fnodes "k8s.io/kubernetes/test/e2e/framework/node"
+	fpod "k8s.io/kubernetes/test/e2e/framework/pod"
+	fpv "k8s.io/kubernetes/test/e2e/framework/pv"
+	k8s "sigs.k8s.io/vsphere-csi-driver/v2/pkg/kubernetes"
+)
+
+// Test to verify RWX volume provision with reclaim policy set and modified
+var _ = ginkgo.Describe("File Volume Test for Reclaim Policy", func() {
+	f := framework.NewDefaultFramework("rwx-tkg-reclaim")
+	var (
+		client            clientset.Interface
+		namespace         string
+		scParameters      map[string]string
+		storagePolicyName string
+		volHealthCheck    bool
+	)
+	ginkgo.BeforeEach(func() {
+		client = f.ClientSet
+		// TODO: Read value from command line
+		volHealthCheck = true
```

Comment:
why do we need this extra variable? by default we need to validate the health check.

Reply:
having a variable will give flexibility for the testers to enable and disable the health check while running in suite.

This was added based on the observation I found while running these tests, the health check is taking approx 20+ min for each pvc (no fine-tuning is done on the RWX tkg setups) - this will increase the overall time required to run the suite for now.

So, the plan is to check in code with health check true and can be configured by tester based on need